pinot-broker/src/main/java/org/apache/pinot/broker/routing/segmentpruner/PartitionSegmentPruner.java

```diff
@@ -61,6 +63,7 @@
   private final ZkHelixPropertyStore<ZNRecord> _propertyStore;
   private final String _segmentZKMetadataPathPrefix;
   private final Map<String, PartitionInfo> _partitionInfoMap = new ConcurrentHashMap<>();
+  private final Map<PartitionInfo, Set<String>> _partitionInfoSegments = new ConcurrentHashMap<>();
```

Comment:
You may probably need to override the hashCode and equals methods in `PartitionInfo` class since the `PartitionInfo` object is used as a key in a hashmap in your PR.

Reply:
Oh, nvm. Saw your methods down below.